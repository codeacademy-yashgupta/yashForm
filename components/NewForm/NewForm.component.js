import React, { Component } from 'react';
import {
  View, StyleSheet, Button,
} from 'react-native';
import style from './NewForm.style';
import Header from '../Header/Header.component';
import CustomTextField from '../CustomTextField/CustomTextField.component';

const styles = StyleSheet.create(style);
export default class NewForm extends Component {
  render() {
    const rows = [];
    const addFieldClick = () => {
      alert('hi');
      rows.push(<CustomTextField />);
    };
    return (
      <View>
        <Header navigation={this.props.navigation} />
        <CustomTextField onChange={() => console.log('hi')} />
        <View
          style={styles.formHorizontalBar}
        />
        <View style={{ marginLeft: '60%' }}>
          <Button title="Add Field" onPress={addFieldClick} />
        </View>
        {rows}
      </View>
    );
  }
}
