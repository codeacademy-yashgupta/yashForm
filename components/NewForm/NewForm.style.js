export default {
  formHorizontalBar: {
    borderBottomColor: 'black', borderBottomWidth: 1, marginTop: 20, marginBottom: 20,
  },
};
