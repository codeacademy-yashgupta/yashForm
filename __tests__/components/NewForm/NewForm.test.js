import React from 'react';
import renderer from 'react-test-renderer';
import NewForm from '../../../components/NewForm/NewForm.component';
import 'react-native';

describe('NewForm', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<NewForm />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
