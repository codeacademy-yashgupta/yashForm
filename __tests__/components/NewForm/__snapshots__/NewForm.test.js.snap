// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewForm should render correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#3700B3",
        "flexDirection": "row",
        "height": 60,
        "marginTop": 20,
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../public/icons/baseline_arrow_back_white_18dp.png",
          }
        }
        style={
          Object {
            "marginTop": 12,
          }
        }
      />
    </View>
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 22,
          "marginLeft": 20,
          "marginTop": 16,
        }
      }
    >
      Header
    </Text>
    <Image
      source={
        Object {
          "testUri": "../../../public/icons/baseline_more_vert_white_48pt_3x.png",
        }
      }
      style={
        Object {
          "height": 47,
          "marginTop": 6,
          "position": "absolute",
          "resizeMode": "contain",
          "right": -50,
        }
      }
    />
  </View>
  <View>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      style={
        Object {
          "borderBottomColor": "black",
          "borderBottomWidth": 1,
          "fontSize": 22,
          "margin": 20,
        }
      }
      underlineColorAndroid="transparent"
    />
  </View>
  <View
    style={
      Object {
        "borderBottomColor": "black",
        "borderBottomWidth": 1,
        "marginBottom": 20,
        "marginTop": 20,
      }
    }
  />
  <View
    style={
      Object {
        "marginLeft": "60%",
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Add Field
        </Text>
      </View>
    </View>
  </View>
  <RCTScrollView>
    <View />
  </RCTScrollView>
  <View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#3700B3",
          "bottom": 0,
          "height": 50,
          "opacity": 1,
          "position": "relative",
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 22,
            "marginTop": 10,
            "textAlign": "center",
          }
        }
      >
        SAVE
      </Text>
    </View>
  </View>
</View>
`;
